app/assets/javascripts/discourse/app/instance-initializers/sidebar.js

```diff
@@ -0,0 +1,27 @@
+import I18n from "I18n";
+import { addSidebarPanel } from "discourse/lib/sidebar/custom-sections";
+
+export default {
+  initialize() {
+    addSidebarPanel((BaseCustomSidebarPanel) => {
```

Comment:
Sorry I'm still not quite sure I understand why we need an initial initialiser here. I think we can just store this object as a default in a constant in the `sidebar/custom-sections.js`. And then when resetting, we can just reset to this default instead of duplicating the initialization of this class in two place.